process/block/preprocess/transactions.go

```diff
@@ -639,10 +741,41 @@ func (txs *transactions) ProcessMiniBlock(miniBlock *block.MiniBlock, haveTime f
 		return err
 	}
 
+	mapHashTx := make(map[string]data.TransactionHandler)
+	for index, txHash := range miniBlockTxHashes {
+		mapHashTx[string(txHash)] = miniBlockTxs[index]
+	}
+
+	gasConsumedByMiniBlockInSenderShard, gasConsumedByMiniBlockInReceiverShard, err := txs.gasHandler.ComputeGasConsumedByMiniBlock(
+		miniBlock,
+		mapHashTx)
+
+	if err != nil {
+		return err
+	}
+
+	var gasConsumedInSelfShard uint64
+	if txs.shardCoordinator.SelfId() == miniBlock.SenderShardID {
```

Comment:
these condition and lines can be extracted into a function.

Reply:
extracted